packages/db/index.js

```diff
@@ -325,6 +325,98 @@ export class DBClient {
     }))
   }
 
+  /**
+   * List a users pinned files.
+   * @param {number} userId
+   * @param {import('./db-client-types').ListUploadsOptions} [opts]
+   * @returns {Promise<Array<import('./db-client-types').UploadItemOutput>>}
+   */
+  async listPinned (userId, opts = {}) {
+    // Query for pins the same shape as the uploads but with !inner joins
+    const pinnedQuery = `
+      _id:id::text,
+      name,
+      created:inserted_at,
+      updated:updated_at,
+      content!inner(
+        cid,
+        dagSize:dag_size,
+        pins:pin!inner(
+          status,
+          updated:updated_at,
+          location:pin_location(
+            _id:id,
+            peerId:peer_id,
+            peerName:peer_name,
+            ipfsPeerId:ipfs_peer_id,
+            region
+          )))`
+    const size = opts?.size || 10
+    // TODO get pinned for all of the users auth keys?
+    const authKeys = await this.listKeys(userId)
+    const authKeyIds = authKeys.map((authKey) => authKey._id)
+
+    let query = this._client
+      .from(psaPinRequestTableName)
+      .select(pinnedQuery)
+      .eq('content.pins.status', 'Pinned')
```

Comment:
I think this will probably suffer from the problem described in #1115. I don't think it's worth trying to fix it here, but maybe just worth a TODO comment here and a note on #1115 to fix this as part of that ticket?